src/Features/Core/Portable/Diagnostics/Analyzers/QualifyMemberAccessDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,120 @@
+
+        private static readonly LocalizableString s_localizableTitleQualifyMembers = new LocalizableResourceString(nameof(FeaturesResources.AddThisOrMeQualification), FeaturesResources.ResourceManager, typeof(FeaturesResources));
+        private static readonly DiagnosticDescriptor s_descriptorQualifyMemberAccess = new DiagnosticDescriptor(IDEDiagnosticIds.AddQualificationDiagnosticId,
+                                                                    s_localizableTitleQualifyMembers,
+                                                                    s_localizableMessage,
+                                                                    DiagnosticCategory.Style,
+                                                                    DiagnosticSeverity.Hidden,
+                                                                    isEnabledByDefault: true,
+                                                                    customTags: DiagnosticCustomTags.Unnecessary);
+
+        private OptionSet _lazyDefaultOptionSet;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_descriptorQualifyMemberAccess);
+
+        protected abstract bool IsAlreadyQualifiedMemberAccess(SyntaxNode node);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterOperationAction(AnalyzeOperation, OperationKind.FieldReferenceExpression, OperationKind.PropertyReferenceExpression, OperationKind.MethodReferenceExpression);
+        }
+
+        private void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var memberReference = (IMemberReferenceExpression)context.Operation;
+
+            // this is a static reference so we don't care if it's qualified
+            if (memberReference.Instance == null)
+            {
+                return;
+            }
+
+            // if we're not referencing `this.` or `Me.` (e.g., a parameter, local, etc.)
+            if (memberReference.Instance.Kind != OperationKind.InstanceReferenceExpression)
+            {
+                return;
+            }
+
+            // if there's already a `this.` or `Me.` qualification
+            if (IsAlreadyQualifiedMemberAccess(memberReference.Instance.Syntax))
+            {
+                return;
+            }
+
+            var optionSet = GetOptionSet(context.Options);
+            if (optionSet == null)
+            {
+                return;
+            }
+
+            var memberSymbol = GetSymbol(memberReference);
+            if (memberSymbol == null)
+            {
+                return;
+            }
+
+            // if the reference is to a base type
+            if (memberSymbol.ContainingSymbol != memberReference.Instance.ResultType)
```

Comment:
`base.` and `MyBase.` references won't ever get this far because they're taken care of on line 43 with `if (memberReference.Instance.Kind != OperationKind.InstanceReferenceExpression) { return }` (and there's a comment there explaining this.)

I've experimented with removing this check (line 71) and we appropriately flag `SomeBaseProperty = 1` as convertable to `Me.SomeBaseProperty = 1`, but since we don't yet have an analyzer/code fix/simplification rule for `base.`/`MyBase.` then you can get into a weird scenario of `MyBase.Property` gets flagged as being able to be simplified to `Property`, but then that's immediately flagged as being able to be qualified with `Me.`.

Ultimately, I'm on the fence about this one and don't care either way.  If you feel strongly about qualifying base access with `Me.` then I can easily remove this check.